logstash-core/lib/logstash/agent.rb

```diff
@@ -106,6 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
+      LogStash::Instrument::Collector.instance.agent = self
```

Comment:
I don't know how can I pass an instance to  the Sinatra App by passing it
to the webserver (aka puma agent here). Only way I know to pass it is
through code, like the collector way, we could add another way. Please let
me know if you know how to pass ref thought webserver.

On Mon, Feb 8, 2016 at 4:13 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In logstash-core/lib/logstash/agent.rb
> <https://github.com/elastic/logstash/pull/4641#discussion_r52177329>:
>
> > @@ -106,6 +106,7 @@ def stop_background_services
> >    def configure_metric
> >      if collect_metric?
> >        @logger.debug("Agent: Configuring metric collection")
> > +      LogStash::Instrument::Collector.instance.agent = self
>
> Do the collector really need to know about the agent? Would it be better
> to start the webserver and give it the reference to the current agent?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4641/files#r52177329>.
>